devtools/dump_devinfo.py

```diff
@@ -343,6 +344,26 @@ def _echo_error(msg: str):
     )
 
 
+def format_exception(e):
+    """Print full exception stack as if it hadn't been caught.
```

Comment:
Is this only to capture the exceptions as a string? If so, using `file` parameter for `traceback.print_stack()` would probably be a better option.

Reply:
`dump_devinfo` was doing `print_stack` which printed the stack from where it was called which didn't actually show the cause of the error.  This function is printing the stack from where the exception is raised (i.e. in `smartprotocol`) and outputs the full stack as if the exception hadn't been caught at all.  I'm not sure if passing `file` achieves the same thing.  What would the `file` parameter be?

Reply:
Ohh, I see. Although I'd have thought that the originating exception would also be there in the exception stack?

The `file` parameter would just allow capturing the print output, as I thought this was just done for the purpose of wrapping it inside `_echo_error`.

Reply:
> Anyways, this looks good to me and it's part of our devtools, so feel free to merge when you want 👍

I split out the `smartprotocol` change into it's own PR https://github.com/python-kasa/python-kasa/pull/884, will merge this as a purely `dump_devinfo` specific PR after that's merged.